app/src/main/java/org/mozilla/fenix/components/metrics/MetricsMiddleware.kt

```diff
@@ -28,6 +28,7 @@ class MetricsMiddleware(
         is AppAction.ResumedMetricsAction -> {
             metrics.track(Event.GrowthData.SetAsDefault)
             metrics.track(Event.GrowthData.FirstWeekSeriesActivity)
+            metrics.track(Event.GrowthData.UsageThreshold)
```

Comment:
> In the second app open, we wouldn't send this data either because now usageTimeGrowthSent is true.

`usageTimeGrowthSent` should not be true until the data is actually sent in a subsequent `onResume` where the `elapsedTimeMillis` has been cached correctly, since it otherwise won't pass the conditional set up in `shouldTrack`. 

I'm open to creating something like an `AppAction.stoppedMetricsAction`, but I'm hesitant because I have a deep distrust of work finishing during destructive lifecycle events (even caching the elapsed time during onPause here feels error prone to me).

In general I am interested in refactoring the underlying design here to be a little more functional for reasons like this - managing this state through a disk cache across multiple calls seems risky. I'm still hoping that if there is continued buy-in for this kind of client-side data processing that we will have an opportunity to engage in a more robust design and implementation cycle.

Reply:
Ah, I missed the other part of the `shouldTrack` check: `settings.usageTimeGrowthData > usageThresholdMillis`

Doesn't that also mean we're only sending the event on the second open in that case. Does that suffice for your use case? If so, this lgtm.

> In general I am interested in refactoring the underlying design here to be a little more functional for reasons like this - managing this state through a disk cache across multiple calls seems risky. I'm still hoping that if there is continued buy-in for this kind of client-side data processing that we will have an opportunity to engage in a more robust design and implementation cycle.

Yes, this should be the right ask eventually after the end of this experiment.

Reply:
> Doesn't that also mean we're only sending the event on the second open in that case. Does that suffice for your use case? If so, this lgtm.

It will mean that yeah, but that suffices for this use case. Thanks for being thorough!